backend/src/middleware/validUrlMiddleware.js

```diff
@@ -0,0 +1,18 @@
+const validURL = str => {
+  const isValid = str.match(/^(?:https?:\/\/)(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
+  return !!isValid
+}
+
+export default {
+  Mutation: {
+    CreateSocialMedia: async (resolve, root, args, context, info) => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T15:36:50Z" title="Friday, April 12th 2019, 5:36:50 pm +02:00">Apr 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,18 @@
+const validURL = str => {
+  const isValid = str.match(/^(?:https?:\/\/)(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
+  return !!isValid
+}
+
+export default {
+  Mutation: {
+    CreateSocialMedia: async (resolve, root, args, context, info) => {
```


@mattwr18 what about `UpdateSocialMedia`? :slightly_smiling_face: Or do we always want to Delete+Create?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:08:00Z" title="Friday, April 12th 2019, 6:08:00 pm +02:00">Apr 12, 2019</time>_

---


I was talking to @Tirokk about this this morning.
The ticket doesn't say anything about updating or deleting social media
So, I would probably create two separate issues for those use cases, which can be separately estimated and potentially would be beginner-friendly tickets for people to work on.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T16:10:44Z" title="Friday, April 12th 2019, 6:10:44 pm +02:00">Apr 12, 2019</time>_

---


@mattwr18 OK cool! That's a solution as well. Just make sure we don't forget these things. :+1:

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:29:04Z" title="Friday, April 12th 2019, 6:29:04 pm +02:00">Apr 12, 2019</time>_

---


I can take responsibility for adding the tickets :smile_cat: